Tests/UnitTests/FileShareClientTests/DownloadFilesTests.cs

```diff
@@ -126,5 +127,69 @@ public async Task TestDownloadFileSetsAuthorizationHeader()
             Assert.AreEqual("bearer", fakeHttpClientFactory.HttpClient.DefaultRequestHeaders.Authorization.Scheme);
             Assert.AreEqual(DUMMY_ACCESS_TOKEN, fakeHttpClientFactory.HttpClient.DefaultRequestHeaders.Authorization.Parameter);
         }
+
+
+        [Test]
+        public async Task TestBasicDownloadFileWithCancellationToken()
+        {
+            var batchId = Guid.NewGuid().ToString();
+            var downldLocation = @"D";
+            var expectedBytes = Encoding.UTF8.GetBytes("Contents of a file.");
+            nextResponse = new MemoryStream(expectedBytes);
+            var result = await fileShareApiClient.DownloadFileAsync(batchId, "AFilename.txt", downldLocation, 10, CancellationToken.None);
+            Assert.IsTrue(result.IsSuccess);
+            Assert.AreEqual($"/basePath/batch/{batchId}/files/AFilename.txt", lastRequestUri.AbsolutePath);
```

Comment:
Can you please verify that 'D' contains the expected bytes?

Reply:
added test scenario when fileSizeInBytes is larger than maxDownloadBytes.

Reply:
added test for the downloaded file contains the expected bytes. Thanks for review.